src/components/users-list/index-list.tsx

```diff
@@ -1,10 +1,30 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import { GetUsers } from '../../domain/users-list';
 import { UserList } from '.';
+import { Button } from '../button';
 
-export const UserListContainer = () => {
+export const UsersPage = () => {
+  const perPage = 5;
+  const [page, setPage] = useState<number>(1);
   const token = localStorage.getItem('token') || undefined;
-  const { loading, error, data } = GetUsers({ token });
+  const { loading, error, data, loadMore } = GetUsers({ token, perPage });
+
+  useEffect(() => {
+    const offset = (page - 1) * perPage;
+    loadMore(offset);
+  }, [page, perPage, loadMore]);
```

Comment:
`useEffect` está se tornando um anti-pattern na comunidade do React, que tal pensar em uma forma de fazer isso sem o uso dele?
https://react.dev/learn/you-might-not-need-an-effect